### DISCUSSION TITLE
Can't get no-restricted-imports to work?

### DISCUSSION THREAD

**Discussion by AnderssonPeter** [eslint/eslint#15729] [Q&A] [answered]
Hi I have the following config
```json
    {
      "files": [
        "*/infrastructure/**/*.ts"
      ],
      "rules": {
        "no-restricted-imports": ["error", {
          "patterns": [{
            "group": ["*/ui/*"], 
            "message": "ui services should not be used in infrastructure!"
          }]
        }]
      }
    }
```
So the idea here is to disallow any import in a file thats in a `infrastructure` folder, that is importing something that's in a `ui` folder.
But it doesn't work in the file `src\app\test\infrastructure\test.infrastructure.module.ts` there is a import of `../ui/test.service` but eslint doesn't yield a error?

What am I doing wrong?

**Comment by awibox** [comment, marked as answer]
Hi! Which version of the eslint is used? Configuration for `no-restricted-imports` is supported from version 7.27.0

**Reply by AnderssonPeter** [reply]
I'm not at work right now but i think I managed to get it to work. Can't remeber how..

**Comment by redplane** [comment]
Same problem with nx project, and it does not work:
![image](https://user-images.githubusercontent.com/7974095/230054378-b9f1ec48-5626-48d1-92d7-7244a2ec749b.png)

![image](https://user-images.githubusercontent.com/7974095/230055026-f39c0d61-e942-4472-ae4b-0646f34ebbd9.png)


If I change to:

```
"no-restricted-imports": ["error", {
          "patterns": [{
            "group": ["api"],
            "message": "ui services should not be used in infrastructure!"
          }]
        }]
```
It will work:
![image](https://user-images.githubusercontent.com/7974095/230054752-d9bb5d01-0e86-4af2-8fb8-64cc95d7cfd8.png)


But I have one more package whose name is: `my-package/api`, the error is thrown there also :|

**Comment by redplane** [comment]
Magically, after asking the above question, I found a solution, hope this helps another people:

```
"no-restricted-imports": ["error", {
          "patterns": [{
            "group": ["**/api/*", "!**/api"],
            "message": "Use @my/app instead"
          }]
        }]
```

Found the above solution by reading this question on [Stackoverflow](https://stackoverflow.com/questions/64995811/eslint-no-restricted-imports-prevent-an-import-from-from-path-ending-with-patt)
